cheroot/workers/monitor.py

```diff
@@ -0,0 +1,155 @@
+    self.cancel(), you'll have to wait until the sleep() call finishes before
+    the thread stops. To compensate, it defaults to being daemonic, which means
+    it won't delay stopping the whole process.
+    """
+
+    def __init__(
+        self,
+        interval,
+        function,
+        args=None,
+        kwargs=None,
+        logger=None,
+    ):
+        """Initialize a BackgroundTask."""
+        super(BackgroundTask, self).__init__()
+        self.interval = interval
+        self.function = function
+        self.args = args if args else []
+        self.kwargs = kwargs if kwargs else {}
+        self.running = False
+        self.log = logger or (lambda msg: None)
+        # default to daemonic
+        self.daemon = True
+
+    def cancel(self):
+        """Cancel the task."""
+        self.running = False
+
+    def run(self):
+        """Run - Invoke function every interval seconds."""
+        self.running = True
+        while self.running:
+            time.sleep(self.interval)
+            if not self.running:
+                return
+            try:
+                self.function(*self.args, **self.kwargs)
+            except Exception as e:
+                self.log(
+                    'Error in background task thread function {}:{}.'.format(
+                        self.function,
+                        e,
+                    ), )
+                # Quit on first error to avoid massive logs.
+                raise
+
+
+class Monitor(object):
+    """Monitor a BackgroundTask."""
+
+    callback = None
+    """The function to call at intervals."""
+
+    frequency = 60
+    """The time in seconds between callback runs."""
+
+    thread = None
+    """A :class:`BackgroundTask` thread.
+    """
+    def __init__(self, callback, frequency=60, name=None, logger=None):
+        """Initialize Monitor."""
+        self.callback = callback
+        self.frequency = frequency
+        self.thread = None
+        self.name = name or self.__class__.__name__
+        self.log = logger or (lambda msg: None)
+
+    def start(self):
+        """Start our callback in its own background thread."""
+        if self.frequency > 0:
+            threadname = self.name
+            if self.thread is None:
+                self.thread = BackgroundTask(
+                    self.frequency,
+                    self.callback,
+                    logger=self.log,
+                )
+                self.thread.setName(threadname)
+                self.thread.start()
+                self.log('Started monitor thread {}.'.format(threadname))
+            else:
+                self.log(
+                    'Monitor thread {} already started.'.format(threadname), )
+
+    start.priority = 70
+
+    def stop(self):
+        """Stop our callback's background task thread."""
+        if self.thread is None:
+            self.log('No thread running for {}.'.format(self.name))
```

Comment:
Interrupting early allows dedenting the following block
```suggestion
            self.log('No thread running for {}.'.format(self.name))
            return
```

Reply:
Done.